fdbserver/VersionedBTree.actor.cpp

```diff
@@ -820,20 +820,21 @@ class FIFOQueue {
 				if (mode == POP) {
 					--queue->numPages;
 				}
-				page.clear();
-				debug_printf("FIFOQueue::Cursor(%s) readNext page exhausted, moved to new page\n", toString().c_str());
 
 				if (mode == POP && !queue->usesExtents) {
 					// Freeing the old page must happen after advancing the cursor and clearing the page reference
 					// because freePage() could cause a push onto a queue that causes a newPageID() call which could
 					// pop() from this very same queue. Queue pages are freed at version 0 because they can be reused
 					// after the next commit.
+					page.clear();
 					queue->pager->freePage(oldPageID, 0);
 				} else if (queue->usesExtents && (p->extentCurPageID == p->extentEndPageID)) {
+					page.clear();
 					// Figure out the beginning of the extent
 					int pagesPerExtent = queue->pagesPerExtent;
 					queue->pager->freeExtent(oldPageID - pagesPerExtent + 1);
 				}
+				debug_printf("FIFOQueue::Cursor(%s) readNext page exhausted, moved to new page\n", toString().c_str());
```

Comment:
`page.clear()` must occur regardless of the `if` / `else if` condtions.  Just move it to after the conditional blocks right before the relocated printf.